airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -175,18 +207,63 @@ def upload_metadata_to_gcs(
     icon_uploaded, icon_blob_id = _icon_upload(metadata, bucket, metadata_file_path)
 
     version_uploaded, version_blob_id = _version_upload(metadata, bucket, metadata_file_path)
+
+    doc_version_uploaded, doc_version_blob_id = _doc_upload(metadata, bucket, metadata_file_path, False, False)
+    doc_inapp_version_uploaded, doc_inapp_version_blob_id = _doc_upload(metadata, bucket, metadata_file_path, False, True)
+    
     if not validator_opts.prerelease_tag:
         latest_uploaded, latest_blob_id = _latest_upload(metadata, bucket, metadata_file_path)
+        doc_latest_uploaded, doc_latest_blob_id = _doc_upload(metadata, bucket, metadata_file_path, True, False)
+        doc_inapp_latest_uploaded, doc_inapp_latest_blob_id = _doc_upload(metadata, bucket, metadata_file_path, True, True)
     else:
         latest_uploaded, latest_blob_id = False, None
+        doc_latest_uploaded, doc_latest_blob_id = doc_inapp_latest_uploaded, doc_inapp_latest_blob_id = False, None
 
     return MetadataUploadInfo(
-        uploaded=version_uploaded or latest_uploaded,
-        latest_uploaded=latest_uploaded,
-        version_uploaded=version_uploaded,
-        version_blob_id=version_blob_id,
-        latest_blob_id=latest_blob_id,
-        icon_blob_id=icon_blob_id,
-        icon_uploaded=icon_uploaded,
+        metadata_uploaded=version_uploaded or latest_uploaded,
         metadata_file_path=str(metadata_file_path),
+        uploaded_files=[
+            UploadedFile(
+                id="version_metadata",
+                uploaded=version_uploaded,
+                description="versioned metadata",
+                blob_id=version_blob_id,
+            ),
+            UploadedFile(
+                id="latest_metadata",
+                uploaded=latest_uploaded,
+                description="latest metadata",
+                blob_id=latest_blob_id,
+            ),
+            UploadedFile(
+                id="icon",
+                uploaded=icon_uploaded,
+                description="icon",
+                blob_id=icon_blob_id,
+            ),
+            UploadedFile(
+                id="doc_version",
+                uploaded=doc_version_uploaded,
+                description="versioned doc",
+                blob_id=doc_version_blob_id,
+            ),
+            UploadedFile(
+                id="doc_latest",
+                uploaded=doc_latest_uploaded,
+                description="latest doc",
+                blob_id=doc_latest_blob_id,
+            ),
+            UploadedFile(
+                id="doc_inapp_version",
+                uploaded=doc_inapp_version_uploaded,
+                description="versioned inapp doc",
+                blob_id=doc_inapp_version_blob_id,
+            ),
+            UploadedFile(
```

Comment:
💅 Ideally we return these from their respective upload functions.

@lmossman Do you think thats a reasonable change we can make?

Reply:
@bnchrch the reason I did it this way was because in some cases we wet the `uploaded` and `blob_id` to `False, None` without calling the upload function. So if `UploadedFile` is returned from the function, then we would need to duplicate the `id` and the `description` in `upload_metadata_to_gcs` as well for those cases where the upload function isn't called, so doing it this way felt DRYer to me.

What do you think?